### PR TITLE
26 doc update documentation for UI setup and development

### PR DESCRIPTION
**Pull request by PlesnikJakub**
## Summary

Documentation enhancement covering contributing, API client, and UI design.

Any fixes or recommendations are welcome. ❤️ 

## Types of changes

- [x] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactoring (non-breaking change to code and structure)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Testing

N/A

## Notes

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
 
-### Scripts
+## Technica stack
+- React + TypeScript
+- AntD 
+
+## Local setup
+
+- run `npm install`
+- Make sure API is running on url specified in .env file
```
Comment (petrspelos) [inline README.md:12]: `doc-link`, `spelling/grammar`

```suggestion
- Make sure [the API](https://github.com/3PillarGlobal-Czechia/interview-app-api) is running on the URL specified in the `.env` file
```

Comment (petrspelos) [reply README.md:12]: `question`

@PlesnikJakub, do you think we should explain how to create and structure the `.env` file?

Or at least what the environment variable name is for the API endpoint? 🤔

File: README.md [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+- run `npm start`
+- application will start on `http://localhost:3000/`
+
+### Other scripts
 
 `npm start` - Runs the app in the development mode \
 `npm test` - Launches the test runner in the interactive watch mode \
-`npm run build` - Builds the app for production to the `build` folder \
-`npm run eject` - If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time
+`npm lint` - Runs eslint with quiet and autofix flag \
```
Comment (petrspelos) [inline README.md:20]: `doc-link`

```suggestion
`npm lint` - Runs [ESLint](https://eslint.org/) with the quiet and autofix flags \
```

File: docs/API_CLIENT.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# API Client
+
+Communication with the backend API is implemented through the client in `ApiClient.ts`. It's an autogenerated file that contains all required methods and types for work with API. 
+
+Auto-generation is done through the package NSwag which uses open API documentation to generate typescript client code. NSwag is listed as a dev dependency in this project.
```
Comment (petrspelos) [inline docs/API_CLIENT.md:5]: `doc-link`

```suggestion
Auto-generation is done through the [NSwag package](https://openbase.com/js/nswag/documentation) which uses the open API documentation to generate typescript client code. NSwag is listed as a dev dependency in this project. 

```

File: docs/API_CLIENT.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+Communication with the backend API is implemented through the client in `ApiClient.ts`. It's an autogenerated file that contains all required methods and types for work with API. 
+
+Auto-generation is done through the package NSwag which uses open API documentation to generate typescript client code. NSwag is listed as a dev dependency in this project. 
+
+`nswag.json` contains required configuration.
+
+## Regeneration
+
+After any changes in the contract on API (new endpoints, URL changes, requests, or return types) ApiClinet needs to be regenerated to reflect all changes.
```
Comment (petrspelos) [inline docs/API_CLIENT.md:11]: `spelling/grammar`

```suggestion
After any changes to the API contract (new endpoints, URL changes, requests, or return types) ApiClient needs to be regenerated to reflect these changes. 

```

File: docs/DESIGN.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# UI/UX Design
+
+## General
+
+The application design is based on UI Toolkit [Ant Design](https://ant.design/) with some custom elements designed specificaly for purposes of this application. The main focus of our design is to make application lightweight, features easily accesible and fast for use.
```
Comment (petrspelos) [inline docs/DESIGN.md:5]: `spelling/grammar`

```suggestion
The application design is based the [Ant Design](https://ant.design/) UI Toolkit with some custom elements designed specifically for the purposes of this application. The main focus of our design is to make the application lightweight, features easily accessible and fast to use.

```

**Review by petrspelos — CHANGES_REQUESTED** [review]
I'd say let's apply at least the `spelling/grammar` suggestions.

- I believe applying the `doc-link` suggestions could preemptively answer the simplest of questions
- The SimpleLogin `optional/suggestion` is certainly something to perhaps discuss

---

Other than that, we've got a set of docs any open source repo can envy us. ☺️ 🎉

File: README.md [hunk 6]
```diff
@@ -1,14 +1,28 @@
 # interview-app-ui
 
-Project descriotions
+Welcome to UI repository of Interview app. Goal of this project is to create a tool that can help technical interviewes to lead an interview with a increased confidence and ability to stay consistent accross of multiple interviews.
```
[the remark this answers is not in the document]
Comment (PlesnikJakub) [reply README.md:3]: No

**Comment by PlesnikJakub** [comment]
I guess I should start marking PRs as drafts to avoid the wall of shame like this 😅

**Comment by petrspelos** [comment]
> I guess I should start marking PRs as drafts to avoid the wall of shame like this 😅

There's no shame in healthy collaboration, @PlesnikJakub. ❤️ #oneteam

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+* Open a new GitHub pull request with the fix.
+
+#### **Do you intend to add a new feature or change an existing one?**
+
+* Open a new GitHub issue with the description of the new feature and add label **suggestion**.
+
+#### **Do you have questions about the source code?**
+
+Contact authors directly via email.
```
[the remark this answers is not in the document]
Comment (PlesnikJakub) [reply CONTRIBUTING.md:27]: We may consider this in the future but for now, I think we are fine. 
Feel free to log this as an issue 😊

**Review by petrspelos — APPROVED** [review]

**Pull request merged**